src/group/mod.rs

```diff
@@ -1,71 +1,175 @@
+use openmls::{
+    prelude::Verifiable,
+    prelude::{
+        group_info::{GroupInfo, VerifiableGroupInfo},
+        CreationFromExternalError, ProposalStore, PublicGroup,
+    },
+    prelude::{
+        ConfirmationTag, LibraryError, OpenMlsSignaturePublicKey, ProcessedMessage,
+        ProcessedMessageContent, Sender, StagedCommit,
+    },
+    treesync::{LeafNode, Node},
+};
+use openmls_rust_crypto::{OpenMlsCryptoProvider, OpenMlsRustCrypto};
 use serde::{Deserialize, Serialize};
-use tls_codec::{TlsDeserialize, TlsSerialize, TlsSize, VLBytes};
 
-use crate::{
-    messages::AssistedMessage,
-    pool::{Ciphersuite, Extension, ProtocolVersion},
-};
+use crate::messages::{AssistedCommit, AssistedGroupInfo, AssistedMessage};
 
-pub mod process;
-mod validate_application;
-mod validate_commit;
-mod validate_proposal;
+use self::errors::ProcessAssistedMessageError;
 
+pub mod errors;
+
+#[derive(Serialize, Deserialize)]
 pub struct Group {
+    public_group: PublicGroup,
     group_info: GroupInfo,
+    #[serde(skip)]
+    backend: OpenMlsRustCrypto,
 }
 
 impl Group {
-    pub fn merge_staged_commit(&mut self, _staged_commit: StagedCommit) {}
-
-    pub fn public_tree() {}
+    /// Create a new group state with the group consisting of the creator's
+    /// leaf.
+    pub fn new(
+        verifiable_group_info: VerifiableGroupInfo,
+        leaf_node: LeafNode,
+    ) -> Result<Self, CreationFromExternalError> {
+        let backend = OpenMlsRustCrypto::default();
+        let nodes = vec![Some(Node::LeafNode(leaf_node.into()))];
+        let (public_group, group_info) = PublicGroup::from_external(
+            &backend,
+            nodes,
+            verifiable_group_info,
+            ProposalStore::default(),
+        )?;
+        Ok(Self {
+            group_info,
+            public_group,
+            backend,
+        })
+    }
 
-    pub fn group_info(&self) -> &GroupInfo {
-        &self.group_info
+    fn backend(&self) -> &OpenMlsRustCrypto {
+        &self.backend
```

Comment:
The name is a tad confusing here

Reply:
Is it? Is there a name you would prefer?

Reply:
Backend is already used elsewhere, maybe something more precise like crypto provider